Looks like we have to leave some of that logic in there since the request appenders are coupled

```
Error: vet: ndb_api/common_helpers.go:143:22: cannot use &PostgresHARequestAppender{} (value of type 
*PostgresHARequestAppender) as RequestAppender value in assignment: *PostgresHARequestAppender does not 
implement RequestAppender (missing method appendCloningRequest)
```

--- a/api/v1alpha1/webhook_suite_test.go
+++ b/api/v1alpha1/webhook_suite_test.go
@@ -641,7 +641,6 @@ func createDefaultClone(metadataName string) *Database {
 				SnapshotId:          DEFAULT_UUID,
 				Profiles:            &(Profiles{}),
 				AdditionalArguments: map[string]string{},
-				IsHighAvailability:  HA,
 			},
 		},
 	}

--- a/controller_adapters/database.go
+++ b/controller_adapters/database.go
@@ -147,7 +147,8 @@ func (d *Database) GetInstanceSize() int {
 
 func (d *Database) GetInstanceIsHighAvailability() bool {
 	if d.IsClone() {
-		return d.Spec.Clone.IsHighAvailability
+		// Clone doesn't currently support HA
+		return false
 	}
 	return d.Spec.Instance.IsHighAvailability
 }

--- a/ndb_api/clone_helpers.go
+++ b/ndb_api/clone_helpers.go
@@ -18,7 +18,6 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/nutanix-cloud-native/ndb-operator/api/v1alpha1"
 	"github.com/nutanix-cloud-native/ndb-operator/common"
 	"github.com/nutanix-cloud-native/ndb-operator/ndb_client"
 	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
@@ -189,6 +188,36 @@ func (a *MongoDbRequestAppender) appendCloningRequest(req *DatabaseCloneRequest,
 func (a *PostgresRequestAppender) appendCloningRequest(req *DatabaseCloneRequest, database DatabaseInterface, reqData map[string]interface{}) (*DatabaseCloneRequest, error) {
 	req.SSHPublicKey = reqData[common.NDB_PARAM_SSH_PUBLIC_KEY].(string)
 	dbPassword := reqData[common.NDB_PARAM_PASSWORD].(string)
+
+	// Default action arguments
+	actionArguments := map[string]string{
+		/* Non-Configurable from additionalArguments*/
+		"vm_name":              database.GetName(),
+		"dbserver_description": "DB Server VM for " + database.GetName(),
+		"db_password":          dbPassword,
+	}
+
+	// Appending/overwriting database actionArguments to actionArguments
+	if err := setConfiguredActionArguments(database, actionArguments); err != nil {
+		return nil, err
+	}
+
+	// Converting action arguments map to list and appending to req.ActionArguments
+	req.ActionArguments = append(req.ActionArguments, convertMapToActionArguments(actionArguments)...)
+
+	// Appending LCMConfig Details if specified
+	if err := appendLCMConfigDetailsToRequest(req, database.GetAdditionalArguments()); err != nil {
+		return nil, err
+	}
+
+	return req, nil
+}
+
+func (a *PostgresHARequestAppender) appendCloningRequest(req *DatabaseCloneRequest, database DatabaseInterface, reqData map[string]interface{}) (*DatabaseCloneRequest, error) {
+	req.SSHPublicKey = reqData[common.NDB_PARAM_SSH_PUBLIC_KEY].(string)
+	dbPassword := reqData[common.NDB_PARAM_PASSWORD].(string)
+
+	req.NodeCount = len(database.GetInstanceNodes())
 
 	// Default action arguments
 	actionArguments := map[string]string{

